### ISSUE TITLE
About pytorch2onnx

### ISSUE THREAD

**Issue by CheungBH** [open-mmlab/mmpose#299]
Hello! I used your resnet18 demo and trained a customized dataset. It works and the trained model can recognize my images. 
Then, I use pytorch2onnx.py to transfer my models into .onnx format. The script worked and the ".onnx" model can generate the same result with the ".pth" one. It seems work.

![resnetonnx](https://user-images.githubusercontent.com/39186017/100060472-8acafc80-2e67-11eb-8ce7-8749e4f78bda.png)

But when I use netron the visualize the model structure, the resnet18 model seems strange: some modules contain some unexpected streams, which is obviously not the structure of resnet18. 
![resnetnetron](https://user-images.githubusercontent.com/39186017/100060166-2871fc00-2e67-11eb-8fd6-83b07296b8ab.png)

I wonder whether it is normal in onnx conversion? Whether it will be wrong in the following conversions like "ncnn"  or "tnn"?

-----------------------------------Edit---------------------------------------
Similarly, when I am converting mobilenet, some unexpected modules appear, though the result is correct. And I also have no idea about the correctness about the model. 
![mobilenetron](https://user-images.githubusercontent.com/39186017/100067241-fbc2e200-2e70-11eb-9682-a51adabd4e88.png)

**Comment by innerlee** [comment]
Could you give more detail about the unexpected modules?

**Comment by CheungBH** [comment]
> Could you give more detail about the unexpected modules?
> 
> edit: A concrete example of the expected, and the unexpected, and why they are different, will be great

(Actually, this issue should be raised in mmclassification lol...)

<img width="461" alt="resnet18" src="https://user-images.githubusercontent.com/39186017/100174677-d59c5100-2f07-11eb-9c79-616a52ac6f87.png">

Well, here is the visualization with netron of Conv5.x layer in resnet18. The lower BasicBlock is the expected one, where the structure is "Conv->BN->Relu->Conv->BN->Relu" sequentially with a shortcut outside. 
However, as for the upper BasicBlock, there is a downsample inside. The downsample should be connected with "Conv->BN->Relu->Conv->BN->Relu" structure sequentially, but from the pic we can see that the upper BasicBlock has two streams, which is different from the original one.

<img width="350" alt="res18cls" src="https://user-images.githubusercontent.com/39186017/100174919-5ce9c480-2f08-11eb-8bb5-8ef250877d89.png">

And also, here's the visualization of the classification layer. The structure should be a "GlobalAdaptivePooling + fc + Softmax" sequentially, which is quite simple. But the model in onnx format has a strange appearance. Two streams take the place of sequential structure and some strange ops like reshape, concat, gather, unsqueeze appear.

But, the outputs of onnx and pytorch format are the same. Here's my confusion about resnet18 onnx.

**Comment by innerlee** [comment]
cc. @RunningLeon

**Comment by RunningLeon** [comment]
@CheungBH  
1. Pls. be noted that there's  a downsampling in conv5_1 with a stride of 2. As it can be referred in the [paper](https://arxiv.org/pdf/1512.03385.pdf). Besides, it is also aligned with Resnet series in torchvision.
2. Converting to ONNX from Pytorch is by tracing and it may produce some unnecessary small ops in the graph. That's why it has to be optimized by tools such as [onnx-simplifier](https://github.com/daquexian/onnx-simplifier).

**Comment by CheungBH** [comment]
oh thank you! I think onnx-simplifier is what exactly I need! 
BTW, I wonder when there is a performance difference before and after using onnx-simplifier?

**Comment by RunningLeon** [comment]
It depends. Normally, it is supposed to give better performance since there are fewer nodes in the graph after simplifying.
